services/QuillLMS/client/app/bundles/Staff/components/evidence/llmPromptDatasets/individualTrial/__tests__/individualTrial.test.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import IndividualTrial from '../individualTrial';
+import { MemoryRouter } from 'react-router-dom';
+
+// Mock EvaluationSection to avoid loading spinner conflicts
+jest.mock('../evaluationSection', () => () => <div>Mock EvaluationSection</div>);
+
+jest.mock('moment', () => {
+  const actualMoment = jest.requireActual('moment');
+  return (...args) => actualMoment.utc(...args); // Use UTC to avoid timezone issues in tests
+});
+
+const mockActivityData = {
+  activity: {
+    id: '1',
+    title: 'Test Activity',
+    prompts: [{ id: '123' }],
+    notes: 'Activity notes',
+  },
+};
+
+const mockDataForTrial = {
+  data: {
+    dataset: {
+      id: 1,
+      version: 2,
+      optimal_count: 3,
+      suboptimal_count: 2,
+    },
+    trial: {
+      id: 1,
+      number: 1,
+      trial_duration: 10,
+      evaluation_duration: 5,
+      test_examples_count: 5,
+      created_at: '2023-01-01T12:00:00Z',
+      vendor: 'OpenAI',
+      llm_version: '1.0',
+      temperature: 0.7,
+      llm_prompt: { name: 'Prompt Test' },
+      status: 'completed',
+      notes: 'Trial notes',
+      results: { confusion_matrix: [[1, 0], [0, 1]] },
+    },
+    histogram: { 0: 5, 1: 3 },
+    g_evals: [
+      {
+        metric: 'G Eval Metric',
+        version: 1,
+      },
+    ],
+    llm_examples: [],
+  },
+};
+
+const queryClient = new QueryClient();
+
+const renderWithRouterAndQueryClient = () => {
+  queryClient.setQueryData(['activity-1', '1'], mockActivityData);
+  queryClient.setQueryData(['dataset-1-trial-1', '1', '1'], mockDataForTrial);
+
+  return render(
+    <QueryClientProvider client={queryClient}>
+      <MemoryRouter>
+        <IndividualTrial match={{ params: { activityId: '1', datasetId: '1', promptConjunction: 'and', trialId: '1' } }} />
+      </MemoryRouter>
+    </QueryClientProvider>
+  );
+};
+
+describe('IndividualTrial', () => {
+  it('loads data', async () => {
+    renderWithRouterAndQueryClient();
+
+    await waitFor(() => expect(screen.queryByAltText('Loading spinner')).not.toBeInTheDocument()); // Disappears when data loads
+
+    expect(screen.getByText('Test Activity')).toBeInTheDocument();
+    expect(screen.getByText('Mock EvaluationSection')).toBeInTheDocument(); // Mocked EvaluationSection
```

Comment:
These are both another opportunity to declare a constant and use it in both set-up and evaluation.  As with other items like this that I've flagged, this is totally a take-or-leave comment since these cases are all straightforward enough that I don't think anyone who looks at this later will be confused.